Source/Extensions/Blazorise.Components/RouterTabs.razor.cs

```diff
@@ -0,0 +1,66 @@
+﻿#region Using directives
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Routing;
+#endregion
+
+namespace Blazorise.Components;
+
+public partial class RouterTabs : ComponentBase, IDisposable
+{
+    private bool disposedValue;
+
+    [CascadingParameter] public RouteData RouteData { get; set; }
+
+    [Inject] public NavigationManager NavigationManager { get; set; }
+    [Inject] public RouterTabsService RouterTabsService { get; set; }
+
+    protected override void OnInitialized()
+    {
+        base.OnInitialized();
+
+        NavigationManager.LocationChanged += OnLocationChanged;
+        RouterTabsService.OnStateHasChanged += OnStateHasChanged;
```

Comment:
Should we check for nulls?

Reply:
We can. But the component would already be in a bad state if these were nulls.